test/patch_test.go

```diff
@@ -0,0 +1,177 @@
+				Expect(err).ToNot(HaveOccurred())
+
+				// Expect updated to match patched resource
+				Expect(updated.Spec).To(Equal(test1.Spec))
+
+				extraTest(og, updated, patched, test1)
+			},
+
+			table.Entry("simple patch",
+				&operator.Installation{
+					TypeMeta:   metav1.TypeMeta{Kind: "Installation", APIVersion: "operator.tigera.io/v1"},
+					ObjectMeta: metav1.ObjectMeta{Name: "test1"},
+					Spec: operator.InstallationSpec{
+						Variant: operator.Calico,
+					},
+				},
+				func(i *operator.Installation) {
+					i.Spec.KubernetesProvider = operator.ProviderEKS
+				}, func(og, upd, ptchd, getted *operator.Installation) {}),
+			table.Entry("sub field update",
+				&operator.Installation{
+					TypeMeta:   metav1.TypeMeta{Kind: "Installation", APIVersion: "operator.tigera.io/v1"},
+					ObjectMeta: metav1.ObjectMeta{Name: "test1"},
+					Spec: operator.InstallationSpec{
+						Variant: operator.Calico,
+						CalicoNetwork: &operator.CalicoNetworkSpec{
+							BGP: operator.BGPOptionPtr(operator.BGPEnabled),
+						},
+					},
+				},
+				func(i *operator.Installation) {
+					i.Spec.KubernetesProvider = operator.ProviderEKS
+					mtu := int32(444)
+					i.Spec.CalicoNetwork.MTU = &mtu
+				}, func(og, upd, ptchd, getted *operator.Installation) {}),
+			table.Entry("IPPool update",
+				&operator.Installation{
+					ObjectMeta: metav1.ObjectMeta{Name: "test1"},
+					Spec: operator.InstallationSpec{
+						CalicoNetwork: &operator.CalicoNetworkSpec{
+							IPPools: []operator.IPPool{
+								{CIDR: "10.0.0.0/24"},
+								{CIDR: "ffee::/24"},
+							},
+						},
+					},
+				},
+				func(i *operator.Installation) {
+					i.Spec.CalicoNetwork.IPPools[0].Encapsulation = operator.EncapsulationIPIPCrossSubnet
+				}, func(og, upd, ptchd, getted *operator.Installation) {
+					pools := getted.Spec.CalicoNetwork.IPPools
+					Expect(pools).To(HaveLen(2))
+					Expect(pools[0].Encapsulation).To(Equal(operator.EncapsulationIPIPCrossSubnet))
+					Expect(pools[1].Encapsulation).To(Equal(operator.EncapsulationType("")))
+				},
+			),
+			table.Entry("Add IPPool",
+				&operator.Installation{
+					ObjectMeta: metav1.ObjectMeta{Name: "test1"},
+					Spec:       operator.InstallationSpec{},
+				},
+				func(i *operator.Installation) {
+					i.Spec.CalicoNetwork = &operator.CalicoNetworkSpec{
+						IPPools: []operator.IPPool{{
+							CIDR:          "10.10.0.0/24",
+							Encapsulation: operator.EncapsulationVXLANCrossSubnet,
+						}},
+					}
+
+				}, func(og, upd, ptchd, getted *operator.Installation) {
+					pools := getted.Spec.CalicoNetwork.IPPools
+					Expect(pools).To(HaveLen(1))
+					Expect(pools[0].CIDR).To(Equal("10.10.0.0/24"))
+					Expect(pools[0].Encapsulation).To(Equal(operator.EncapsulationVXLANCrossSubnet))
+				},
+			),
+		)
+	})
+})
+
+func cleanInstallationResources(c client.Client) error {
+	ctx := context.Background()
+	instList := &operator.InstallationList{}
+	err := c.List(ctx, instList)
```

Comment:
just curious, what made you go for the `fv` test instead of a ut with fake clients? you know how much i like fake clients :)

Reply:
My recollection was that the fake clients don't handle Patching the resource. I'll try with a fake client though.